contrib/endpoints/include/api_manager/api_manager.h

```diff
@@ -87,6 +87,12 @@ class ApiManager {
   virtual utils::Status GetStatistics(
       ApiManagerStatistics *statistics) const = 0;
 
+  // Returns the ApiManager initialization status.
+  //  - Code::UNKNOWN   Not initialized yet. The default value.
+  //  - Code::OK        Successfully initialized
+  //  - Code::ABORTED   Initialization was failed
+  virtual utils::Status InitializationStatus() = 0;
```

Comment:
nginx module is checking the status.
While the ApiManager is waiting for the initialization of the ConfigManager, the status is UNKNOWN and nginx posts the event to handle it later.
If the status is ABORTED, that means ApiManager was failed to load service configs. module returns NGX_DECLINED.
If it is OK, then  ctx->request_handler->Check will be called.

Before the initialization, service_context_map_ is empty.